src/cli/serve/serve.js

```diff
@@ -195,7 +195,11 @@ export default function(program) {
       []
     )
     .option('--plugins <path>', 'an alias for --plugin-dir', pluginDirCollector)
-    .option('--optimize', 'Run the legacy plugin optimizer and then stop the server');
+    .option('--optimize', 'Run the legacy plugin optimizer and then stop the server')
+    .option(
+      '--dry-run-migration',
+      'Perform a dry run saved objects migration, report the results and exit'
+    );
```

Comment:
Do we really want that kind of utility command to be accessible from the `serve` script? My feeling is that we are going to have more of more options that perform one-shot actions without starting the server (I'm thinking of the archiver data migration for example if/when we will do it)

I know this will probably leads to heavy duplication (or factorization), but maybe it's time to add a new 'toolbox' script to avoid polluting `serve` ?

Reply:
Is your concern that the dry run cli is the same executable (`bin/kibana` in a distributable) as running Kibana or is it that we're putting too much functionality into the serve script.

There's a lot of overlap (coupling) between dry run migrations and running kibana server because dry run migrations start the full server. To perform the dry run migration we need to start core with all the plugins loaded which means we need to potentially support some of the plugin discovery flags like `--plugin-dir` `--plugin-path`.

It could also be useful to be able to override some config values with command line arguments instead of requiring everything to be in `kibana.yml`.

Reply:
> Is your concern that the dry run cli is the same executable (bin/kibana in a distributable) as running Kibana or is it that we're putting too much functionality into the serve script.

Both actually, but mostly that it's the same executable.